src/dialects/db2/query-generator-typescript.ts

```diff
@@ -0,0 +1,43 @@
+import { noSchemaParameter } from '../../utils/deprecations';
+import { joinSQLFragments } from '../../utils/join-sql-fragments';
+import { AbstractQueryGenerator } from '../abstract/query-generator';
+import type { TableName } from '../abstract/query-interface';
+
+/**
+ * Temporary class to ease the TypeScript migration
+ */
+export class Db2QueryGeneratorTypeScript extends AbstractQueryGenerator {
+  // TODO [>7]: remove schema parameter
+  describeTableQuery(tableName: TableName, schema?: string) {
```

Comment:
Can you write a quick test where you use `sequelize.sync({ schema: 'something' })`? (possibly with alter too, since describeTableQuery probably only runs if the table already exists)

That should be supported and should not print a deprecation warning, because internally `sequelize.sync` should modify the first parameter, instead of providing the second one

Reply:
Not sure what test you have in mind, something in the `QueryInterface#describeTable` tests?

Reply:
I'm not sure why I said test. When I was writing thing I was thinking testing for this would be difficult but running a quick piece of code and checking the console's logs manually would be fine

Reply:
It's printing the deprecation warning, but maybe something to fix in #14687 ?

Reply:
It's not related to #14687, `Model.sync` is passing that option to `queryInterface.describeTable`

It's best if the PR that adds the deprecation notice fixes that at the same time, so I'd say either we move the deprecation notice to another PR + tweak `Model.sync`, or we do both here

Reply:
I'll look into tweaking it here then

Reply:
Woops, I was wrong. We seem to be in the clear